src/Microsoft.Azure.SignalR.Common/RestClients/HttpExceptionWrapperExtension.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using Microsoft.Rest;
+
+namespace Microsoft.Azure.SignalR.Common.RestClients
+{
+    internal static class HttpExceptionWrapperExtension
+    {
+        internal static Exception WrapAsAzureSingalRException(this Exception e, Uri baseUri)
+        {
+            switch (e)
+            {
+                case HttpOperationException operationException:
+                    var response = operationException.Response;
+                    var request = operationException.Request;
+                    var detail = response.Content;
```

Comment:
what would be inside `detail`? what if the server returns for example: `StatusCode(status, new { detail = "This is unhealthy" }`

Reply:
This is consistent with old REST client and it tries to give more information to users. But if server doesn't give more information other than status code, I don't think we can do more things.

Reply:
server will provide more details as similar to `StatusCode(status, new { detail = "This is unhealthy" }` later, just want to confirm if in that case `detail` would work

Reply:
`var request = operationException.Request; `  The `request` here is a copy of the origin request, so the `detail` is the response content.  In the case you mentioned it works.